modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaCmsTelemetry.php

```diff
@@ -251,16 +278,86 @@ private function shouldSendTelemetryData(): bool {
     if ($isCI) {
       return FALSE;
     }
-    if (AcquiaDrupalEnvironmentDetector::isAhEnv()) {
-      $telemetryOpted = $this->state->get('acquia_connector.telemetry.opted', TRUE);
-      $isAcquiaTelemetryDataSent = $this->state->get('acquia_cms_telemetry.status', FALSE);
-      if ($telemetryOpted && !$isAcquiaTelemetryDataSent) {
-        return TRUE;
-      }
+    $sendTimestamp = $this->state->get('acquia_cms_common.telemetry.timestamp');
+    if (AcquiaDrupalEnvironmentDetector::isAhEnv() &&
+    $this->state->get('acquia_connector.telemetry.opted', TRUE) &&
+    $this->time->getCurrentTime() - $sendTimestamp > 86400 &&
+    $this->shouldResendData()) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Decides if telemetry data should resend or not.
+   */
+  private function shouldResendData(): bool {
```

Comment:
@rajeshreeputra There should be PHPUnit tests for this function as well.

Reply:
it is added [here](https://github.com/acquia/acquia_cms/pull/1615/files#diff-a2c7c381c6440893407a9356e10a6267241a809af9ac6b855be5a206466d88c7R96-R107), but will segregate and update it as separate method.